version/check.go

```diff
@@ -0,0 +1,93 @@
+package version
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+)
+
+const versionTimeout = time.Second * 1
+
+type Version struct {
+	latestVersion *string
```

Comment:
this should be of type string not pointer to it.
Also lets rename this to `number` or something instead of latestVersion, basically saying `version.number`.
Ok with calling it `name` as well.
@ravisuhag ?

Reply:
name is better. as the number will assume it is an integer type. WDYT?

Reply:
yeah I was having same thoughts around the data type, but we usually use version number hence thought to check.

@scortier Let's go with `name`. 👍